engine/table/src/main/java/io/deephaven/engine/table/impl/sources/BooleanSparseArraySource.java

```diff
@@ -106,7 +106,7 @@ public void setNull(long key) {
             return;
         }
         final int indexWithinBlock = (int) (key & INDEX_MASK);
-        if (blocks2[indexWithinBlock] == NULL_BOOLEAN_AS_BYTE) {
+        if (BooleanUtils.isNull(blocks2[indexWithinBlock])) {
```

Comment:
So this is an interesting thing, in that we are explicitly deciding not to normalize the values.  Which is maybe totally fine if we assert the engine should follow the BooleanUtils.isNull definition.  Reading into a chunk won't enforce that though, so there is maybe some danger here.

Reply:
My logic here was that while I could clearly break barrage and parquet, I couldn't see for other cases a way to be sure that non-standard values couldnt creep in. If we can be confident that other values shouldnt be possible here, we can assert that, and only check the three "real" values?

Reply:
Yeah, so my thought regarding this one is that if everything else is right; this is right too.  If there is another place in the code that is wrong, then this is going to exacerbate it compared to the prior behavior which would say "oh you have some wrong null in here, let me set it to the right null".